core/trino-main/src/main/java/io/trino/execution/buffer/PageDeserializer.java

```diff
@@ -427,6 +431,16 @@ private static int getCompressedBlockSize(int compressedBlockMarker)
             return compressedBlockMarker & ~SERIALIZED_PAGE_COMPRESSED_BLOCK_MASK;
         }
 
+        private static int maxCompressedLength(Decompressor decompressor, int blockSizeInBytes)
+        {
+            return switch (decompressor) {
```

Comment:
This is silly... we should add `maxCompressedLength()` to `Decompressor`

Reply:
Yes, that was super strange and I think unnecessary.

The maxCompressedLength is always called with the const block size which means that it never changes and can be precomputed to avoid allocating new compressor every time.